Foundation/NSRunLoop.swift

```diff
@@ -95,10 +95,6 @@ extension NSRunLoop {
             return false
         }
         let modeArg = mode._cfObject
-        if _CFRunLoopFinished(_cfRunLoop, modeArg) {
-            return false
-        }
-        
```

Comment:
_CFRunLoopFinished is returning correctly in the previous case since there is no port or any source scheduled on the run loop. If there is nothing to keep the loop alive, the loop is finished.